sdk/core/src/retry_policy.rs

```diff
@@ -0,0 +1,123 @@
+use tokio::time::{sleep, Duration};
+
+#[async_trait::async_trait]
+pub trait RetryPolicy {
+    async fn create_instance(&self) -> Box<dyn RetryPolicyInstance>;
```

Comment:
Is there a reason `create_instance` is async?

Reply:
In our case no but since clients could implement `RetryPolicy` I think it's a good idea to allow them to use async calls. For example you can have a `RetryPolicy` that "resets" a connection (for example to a db) and it could definitely require an async call.

Reply:
But wouldn't that resetting of the connection happen in the `retry` method of `RetryPolicyInstance`?

Reply:
Yes it could happen there. I would have preferred to separate the initialization phase of the policy instance from its execution (like a constructor), but I suppose I would have to make it fallible as well.

Let's make it simple and keep it infallible and sync! :+1: